### PR TITLE
Convert `cards` dashboard actions to TypeScript

### PR DESCRIPTION
**Pull request by kulyk**
Converts `metabase/dashboard/actions/cards` actions file to TypeScript, no functional changes are expected
**To verify:** CI should be green

### CODE REVIEW

File: frontend/src/metabase/dashboard/actions/cards.js
Comment (kulyk) [inline frontend/src/metabase/dashboard/actions/cards.js:1]: Only keeping `addActionToDashboard` in JavaScript for now. It's awkward, but we can't remember if `card` on `ActionDashboardCard` is a model its saved to or a virtual card with `display: "action"`. We can followup on that one later

File: frontend/src/metabase-types/api/dashboard.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export type DashCardId = number;
 export type BaseDashboardCard = {
   id: DashCardId;
   dashboard_id: DashboardId;
-  dashboard_tab_id?: DashboardTabId;
+  dashboard_tab_id?: DashboardTabId | null;
```
Comment (alxnddr) [inline frontend/src/metabase-types/api/dashboard.ts:57]: Can we narrow it down to `dashboard_tab_id: DashboardTabId | null;`?

**Review by alxnddr — APPROVED** [review]
Looks good! I left one suggestion, please have a look before merging

File: frontend/src/metabase-types/api/dashboard.ts (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase-types/api/dashboard.ts]: Good point 👍 Fixed in [`2ba9048`](https://github.com/metabase/metabase/pull/38447/commits/2ba90489dc1b3d1617306b18fe12c1d9a6537703)

**Pull request merged**